src/server.js

```diff
@@ -1,48 +1,133 @@
+// Initialize Server
+const initialize = async () => {
   logger.info(`Server listen on port ${SERVER_PORT}...`)
-  logger.info(`Entering in ${STATE_START_SERVER} state...`)
   await db()
+  await game.insert({ state: GAME_WAITING })
+}
+
+// Update game state into DB
+const setGameState = async (state) => {
+  const [s] = await game.getAll()
+  await game.update({ ...s, state })
+}
+
+// Run the server
+const run = () => {
+  // Send game state to all clients when it changes
+  game.onChange(s => io.emit(events.SERVER_SET_STATE, s.state))
+  // Set game state to 'WAITING'
+  setGameState(GAME_WAITING)
+
+  // When a client is connected
+  io.on(events.CONNECTION, async (socket) => {
+    logger.info(`Client connected : ${socket.id}`)
 
-  init()
-  setInterval(physicLoop, (1 / 60) * 1000)
+    // Send game state when just connected
+    const [s] = await game.getAll()
+    socket.emit(events.SERVER_SET_STATE, s.state)
+    socket.emit(events.SERVER_ADD_PLAYERS, await player.getAll())
 
-  io.on(CONNECTION, async (socket) => {
-    const newPlayer = {
-      name: NAMES[rand(0, NAMES.length - 1)],
-      x: randPos(1, 7),
-      y: randPos(1, 6),
-    }
+    // When new player is connected
+    socket.on(events.CLIENT_PLAYER_CONNECT, async ({ name }) => {
+      logger.info(`Player connected ''${name}'': ${socket.id}`)
 
-    player.insert(newPlayer)
-    addBodies(createBox(newPlayer.x, newPlayer.y))
-    start = true
+      // Store new player and with its socket ID
+      const newPlayer = {
+        id: socket.id,
+        name,
+        color: utils.randomColor(),
+        x: utils.random(1, 7),
+        y: utils.random(1, 6),
+        latency: 0,
+      }
+      await player.insert(newPlayer)
+      socket.emit(events.SERVER_SET_PLAYER, socket.id)
 
-    // Connect events
-    socket.emit(SET_PLAYERS, await player.getAll())
-    player.onChange(p => io.emit(UPDATE_PLAYER, p))
-    socket.on(UPDATE_PLAYER, player.update) // Should be checked (for cheating player)
+      // Send players to all clients
+      io.emit(events.SERVER_ADD_PLAYERS, await player.getAll())
+    })
+
+    // When the game is started
+    socket.on(events.CLIENT_START_GAME, async () => {
+      // Re-init values if restarted
+      players = []
+      if (physicInterval) clearInterval(physicInterval)
+      if (synchroInterval) clearInterval(synchroInterval)
+
+      // Initialize physic server
+      physic.init()
+      physicInterval = setInterval(physic.tick, (1 / 60) * 1000)
+
+      // Add players to physic engine
+      const playersConnected = await player.getAll()
+      playersConnected.forEach(p => players.push({ ...p, body: physic.createBox(p.x, p.y) }))
+
+      // Set server state to 'STARTED'
+      setGameState(GAME_STARTED)
+
+      // Synchronize clients with server physics
+      synchroInterval = setInterval(async () => {
+        io.emit(events.SERVER_SYNCHRONIZE, players.map(p => ({
+          id: p.id,
+          force: p.body.force,
```

Comment:
lodash/pick

use an helper function and use it in both `server.js` and `game.js`